scene/gui/control.cpp

```diff
@@ -446,9 +446,35 @@ void Control::_validate_property(PropertyInfo &p_property) const {
 	if (p_property.name == "theme_type_variation") {
 		List<StringName> names;
 
-		// Only the default theme and the project theme are used for the list of options.
-		// This is an imposed limitation to simplify the logic needed to leverage those options.
 		ThemeDB::get_singleton()->get_default_theme()->get_type_variation_list(get_class_name(), &names);
+
+		//iterate to find all themes
+		//I don't want to mess with the code too much. this needs improvements.
+		Control *tmp_control = Object::cast_to<Control>(get_parent());
+		Window *tmp_window = Object::cast_to<Window>(get_parent());
+		while (tmp_control || tmp_window) {
+			//we go up and any non Control/Window will break the chain
+			if (tmp_control) {
+				if (tmp_control->get_theme().is_valid()) {
+					tmp_control->get_theme()->get_type_variation_list(get_class_name(), &names);
+				}
+				Control *sub_control = Object::cast_to<Control>(tmp_control->get_parent());
+				tmp_window = Object::cast_to<Window>(tmp_control->get_parent());
+				tmp_control = sub_control;
+			} else {
+				//tmp_window
+				if (tmp_window->get_theme().is_valid()) {
+					tmp_window->get_theme()->get_type_variation_list(get_class_name(), &names);
+				}
+				Window *sub_window = Object::cast_to<Window>(tmp_window->get_parent());
+				tmp_control = Object::cast_to<Control>(tmp_window->get_parent());
+				tmp_window = sub_window;
```

Comment:
Isn't it the same as
```suggestion
				tmp_control = Object::cast_to<Control>(tmp_window->get_parent());
				tmp_window = Object::cast_to<Window>(tmp_window->get_parent());
```
?
Same above.

Reply:
Yes. I don't code in C++ very often so I'm a bit afraid of pointers.